src/ConfigReader.ts

```diff
@@ -0,0 +1,58 @@
+﻿import toml = require("toml");
+import fs = require('fs');
+import { logger } from "./LoggerWrapper";
+import { SettingsRoot } from "./ConfigTypes"
+
+export class ConfigReader {
+    private readonly configPath = "/mnt/settings/application-monitoring-settings";
+
+    public static async ReadConfig(file?: string): Promise<AddonConfig> {
+        let instance: ConfigReader;
+        instance = new ConfigReader();
+        const filePath: string = file!=null ? file : instance.configPath;
+        let config: AddonConfig = new AddonConfig(null);
+        return new Promise<AddonConfig>((resolve, rejectfs) => {
+            logger.info(`attermpting to read ${filePath}`);
+
+            fs.readFile(file, (err, data) => {
+                if (!err) {
+                    const fileContent = data.toString();
+                    logger.info(`read config content ${fileContent}`);
+                    const parsedContent = toml.parse(fileContent)
+                    config = new AddonConfig(parsedContent);
+                } else {
+                    logger.info(`config does not exist, using defaults`);
+                }
+                logger.info(`excluded namespaces ${config.excludedNamespaces}`);
+                resolve(config);
+            })
+        }).catch(err => {
+            logger.error(`${err}`);
```

Comment:
Will we see this output anywhere?

Reply:
yes you see it at the console for the moment, just like we see all the other logs in the pod